README.md

```diff
@@ -1,196 +1,289 @@
+**Note** DNAnexus login is persistent and the login environment is stored on disk in the the Ansible user's home directory. User of this playbook responsibility to make sure that every Ansible user (`monitored_user`) with a streaming upload job assigned has been logged into DNAnexus by either specifying a `dx_token` or `dx_user_token`.
+#### Example Playbook
+`dx-upload-play.yml`
+```YAML
+---
+- hosts: localhost
+  vars:
+    monitored_users:
+      - username: travis
+        local_tar_directory: ~/new_location/upload/TMP
+        local_log_directory: ~/another_location/upload/LOG
+        monitored_directories:
+          - ~/runs
+        applet: applet-Bq2Kkgj08FqbjV3J8xJ0K3gG
+        downstream_input: '{"sequencing_center": "CENTER_A"}'
+      - username: root
+        monitored_directories:
+          - ~/home/root/runs
+        workflow: workflow-BvFz31j0Y7V5QPf09x9y91pF
+        downstream_input: '{"0.sequencing_center: "CENTER_A"}'
+    mode: debug
+    upload_project: project-BpyQyjj0Y7V0Gbg7g52Pqf8q
+  roles:
+    - dx-streaming-upload
+```
+**Note**: For security reasons, you should refrain from storing the DNAnexus authentication token in a playbook that is open-access. One might trigger the playbook on the command line with extra-vars to supply the necessary authentication token, or store them in a closed-source yaml variable file.
+ie. `ansible-playbook dx-upload-play.yml -i inventory --extra-vars "dx_token=<SECRET_TOKEN>"`
+We recommend that the token given is limited in scope to the upload project, and has no higher than **CONTRIBUTE** privileges.
+#### Example Script
+The following is an example script that writes a flat file to the RUN directory once a RUN directory has been successfully streamed.
+Recall that the script will be triggered with a single command line parameter, where `$1` is the path to the local RUN directory  that has been successfully streamed to DNAnexus.
+```
+#!/bin/bash
+set -e -x -o pipefail
+rundir="$1"
+echo "Completed streaming run directory: $rundir" > "$rundir/COMPLETE.txt"
+```
+#### Actions performed by Role
+The dx-streaming-upload role perform, broadly, the following:
+1. Installs the DNAnexus tools [dx-toolkit](https://wiki.dnanexus.com/Downloads#DNAnexus-Platform-SDK) and [upload agent](https://wiki.dnanexus.com/Downloads#Upload-Agent) on the local machine.
+2. Set up a CRON job that monitors a given directory for RUN directories periodically, and streams the RUN directory into a DNAnexus project, triggering an app(let)/workflow upon successful upload of the directory and a local script (when specified by user)

```

Comment:
--> " 2. Sets up a CRON job that monitors a given directory for RUN directories periodically and streams the RUN directory into a DNAnexus project, triggering an app(let)/workflow upon successful upload of the directory and a local script (when specified by the user)."